model-optimizer/mo/ops/strided_slice.py

```diff
@@ -126,21 +126,27 @@ def infer(node: Node):
         node.ellipsis_mask[idx] = 0
 
         if node.graph.graph['layout'] == 'NHWC' and node.out_port(0).data.get_value() is None:
-            PermuteAttrs.create_permute_attrs(node, attrs=[('shrink_axis_mask', 'input:0', permute_masks),
-                                                           ('new_axis_mask', 'input:0', permute_masks),
-                                                           ('ellipsis_mask', 'input:0', permute_masks),
-                                                           ('begin_mask', 'input:0', permute_masks),
-                                                           ('end_mask', 'input:0', permute_masks),
-                                                           ])
-            # permute inputs
             in_shape = node.in_port(0).get_source().data.get_shape()
             assert in_shape is not None, \
                 'Input shape is unknown for 0 input of node {}'.format(node.name)
             input_rank = len(in_shape)
-            if input_rank > 3:
+            output_rank = len(out_shape)
+            if input_rank > 3 and output_rank > 3:
+                # case of to execute StridedSlice in NCHW mode
+                # permute inputs and attributes
+                PermuteAttrs.create_permute_attrs(node, attrs=[('shrink_axis_mask', 'input:0', permute_masks),
+                                                               ('new_axis_mask', 'input:0', permute_masks),
+                                                               ('ellipsis_mask', 'input:0', permute_masks),
+                                                               ('begin_mask', 'input:0', permute_masks),
+                                                               ('end_mask', 'input:0', permute_masks),
+                                                               ])
                 for i_port in node.in_ports().values():
                     if i_port.idx == 0 or i_port.disconnected():
                         continue
                     new_value = permute_array(node, i_port.data.get_value())
                     # set_value additionally set_shape and propagate value to Const node
                     i_port.data.set_value(new_value)
+            else:
+                # case of to execute StridedSlice in NHWC mode
+                # and no need to inject Transpose operation before StridedSlice and after
+                node['reinterp_shape'] = True
```

Comment:
I don't understand. The `reinterp_shape` attribute is set when we want to insert the Transpose operation, but the comment states the opposite.

Reply:
Corrected the code and made a comment.